scipy/optimize/_direct.py

```diff
@@ -0,0 +1,203 @@
+        Boolean flag indicating if the optimizer exited successfully and
+        ``message`` which describes the cause of the termination. See
+        `OptimizeResult` for a description of other attributes.
+
+    Notes
+    -----
+
+    DIRECT is a deterministic
+    optimization algorithm capable of minimizing a black box function with
+    its variables subject to lower and upper bound constrains by sampling
+    potential solutions in the search space. The algorithm starts by
+    normalising the hyperrectangle (set of all possible values that can
+    be taken by the input variables subject to the bound constraints) to
+    an n-dimensional unit hypercube. It samples the function at the
+    center of this hypercube and at 2n (n is the number of variables)
+    more points, 2 in each coordinate direction. Using these function
+    values, DIRECT then divides the domain into hyperrectangles, each
+    having exactly one of the sampling points as its center. In each
+    iteration, DIRECT chooses, using the epsilon parameter, which defaults
+    to 1e-4, some of the existing hyperrectangles to be further divided.
+    This division process continues until the maximum iterations or
+    maximum function evaluations allowed are exceeded, or the function
+    value is within the desired percentage error of the global minimum
+    (if known). The locally biased variant of DIRECT
+    (originally called DIRECT_L) [2]_ is used by default. It makes the
+    search more locally biased and more efficient for cases with only a
+    few local minima.
+
+    This code is based on the DIRECT 2.0.4 Fortran code by Gablonsky et al. at
+    http://www4.ncsu.edu/~ctk/SOFTWARE/DIRECTv204.tar.gz
+    The C version was initially converted via f2c and then cleaned up and
+    reorganized by Steven G. Johnson, August 2007. And this method wraps
+    the C implementation.
+
+    .. versionadded:: 1.8.0
+
+    References
+    ----------
+    .. [1] Jones, D.R., Perttunen, C.D. & Stuckman, B.E. Lipschitzian
+           optimization without the Lipschitz constant. J Optim Theory Appl
+           79, 157-181 (1993)
+    .. [2] Gablonsky, J., Kelley, C. A Locally-Biased form of the DIRECT
+           Algorithm. Journal of Global Optimization 21, 27-37 (2001).
+    """
+    if not isinstance(bounds, Bounds):
+        lb, ub = old_bound_to_new(bounds)
+        bounds = Bounds(lb, ub)
+
+    lb = np.ascontiguousarray(bounds.lb)
+    ub = np.ascontiguousarray(bounds.ub)
+
+    def _func_wrap(x, *args):
+        x = np.asarray(x)
+        f = func(x, *args)
+        return f
+
+    x, fun, ret_code, nfev, nit = _direct(
+        _func_wrap,
+        np.asarray(lb), np.asarray(ub),
+        args,
+        disp, iatol, maxfun, maxiter,
+        locally_biased,
+        f_min, f_min_per,
+        vol_per, sigma_per, callback
+    )
+
+    format_val = (maxfun, maxiter, f_min_per, vol_per, vol_per)
+    if ret_code > 2:
+        message = SUCCESS_MESSAGES[ret_code - 1].format(
```

Comment:
```suggestion
message = SUCCESS_MESSAGES[ret_code - 3].format(
```
There are only 3 success messages and success return codes are 3, 4 and 5. Found this while benchmarking the termination criteria.